app/controllers/ingredient_transfers_controller.rb

```diff
@@ -0,0 +1,67 @@
+class IngredientTransfersController < ApplicationController
+
+def new
+  @source_character = Character.find_by(slug: params[:character_id])
+  @source_inventory = Inventory.where(character_id: @source_character.id)
+  @source_ingredients = []
+  @source_inventory.each do |ingredient|
+   @source_ingredients<<  ingredient.ingredient
+  end
+  @characters = Character.where(archived:false)
+  @ingredient_transfer = IngredientTransfer.new
+end
+
+def create
+  source_character = character = Character.find_by(slug: params[:character_id])
+  ingredient = Ingredient.find_by_id(params["ingredient_transfer"]["ingredient"].to_i)
+  source_inventory = Inventory.where("character_id = ? AND ingredient_id = ?", source_character.id, ingredient.id).first
```

Comment:
You can specify multiple where conditions in a hash, e.g. `Inventory.where(character: source_character, ingredient: ingredient).first` &ndash; which is equivalent to `Inventory.find_by(character: source_character, ingredient: ingredient)`.